File: quests/qey2hh1/Guard_Gregor.pl
sub EVENT_COMBAT {
	if($combat_state == 1){
	my $cur_target = $npc->GetHateTop();
		if($cur_target) {
		my $target_name = $cur_target->GetCleanName();
		quest::say("Time to die $target_name!");
		}
	}
}

sub EVENT_SAY {
  if(($ulevel >= 20) && ($ulevel <= 30)) {
    if($text=~/hail/i) {
      quest::say("Oh, hello $name. Would you like to assist me in a [task]?");
    }
    if($text=~/task/i) {
      quest::say("Excellent! I was hoping you would be able to help me.");
      quest::taskselector(143); #Task: Bringing Goods to the Karanas
    }
  }
  else {
    quest::say("Sorry, $name. I am quite busy...");
  }
}

sub EVENT_ITEM {
  plugin::return_items(\%itemcount);
}

File: quests/qey2hh1/Guard_Gregor.lua
function event_combat(e)
	if(e.joined == true) then
		e.self:Say(string.format("Time to die %s!",e.other:GetCleanName()));
	end
end

function event_say(e)
	if(e.other:GetLevel() >= 20 and e.other:GetLevel() <= 30) then
		if(e.message:findi("hail")) then
			e.self:Say(string.format("Oh, hello %s. Would you like to assist me in a [task]?",e.other:GetName()));
		elseif(e.message:findi("task")) then
			e.self:Say("Excellent! I was hoping you would be able to help me.");
			eq.task_selector(143); -- Task: Bringing Goods to the Karanas
		end
	else
		e.self:Say(string.format("Sorry, %s. I am quite busy...",e.other:GetName()));
	end
end

function event_trade(e)
	local item_lib = require("items");
	item_lib.return_items(e.self, e.other, e.trade)
end